src/entities/Snapshot/utils.ts

```diff
@@ -54,6 +54,32 @@ export function calculateMatch(votes1: SnapshotVote[] | null, votes2: SnapshotVo
   return match
 }
 
+export type OutcomeMatch = {
+  outcomeMatch: number
+  totalProposals: number
+}
+
+export function outcomeMatch(votes: SnapshotVote[]): OutcomeMatch {
+  let matchCounter = 0
+  let closedProposalsCounter = 0
+  for (const vote of votes) {
+    if (vote.proposal?.state === 'closed' && vote.proposal?.scores && vote.proposal?.scores.length > 0) {
+      closedProposalsCounter++
+      const scores = vote.proposal.scores
+      const winnerChoiceScore = Math.max(...scores)
+      const winnerChoiceIndex = scores.indexOf(winnerChoiceScore)
+      const resultIsNotATie = scores.filter((score) => score === winnerChoiceScore).length === 1
+      if (resultIsNotATie) {
+        if (vote.choice === winnerChoiceIndex + 1) {
```

Comment:
I think this can be in one condition

`if (resultIsNotATie && vote.choice === winnerChoiceIndex + 1) {`